### PR TITLE
Change beacon state

### PR DESCRIPTION
**Pull request by cemozerr**
## PR Description
- Changes BeaconState.java to suit 0.5.1
- Adds BeaconBlockHeader data struct

### CODE REVIEW

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/Constants.java [hunk 1]
```diff
@@ -48,7 +48,9 @@
   public static int MIN_SEED_LOOKAHEAD = 1; // 2^0 epochs (6.4 minutes)
   public static int ACTIVATION_EXIT_DELAY = 4; // 2^2 epochs (25.6 minutes)
   public static int EPOCHS_PER_ETH1_VOTING_PERIOD = 16; // 2^4 epochs (~1.7 hours)
+  public static int SLOTS_PER_HISTORICAL_ROOT = 8192; // 2^4 epochs (~1.7 hours)
```
Comment (schroedingerscode) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/Constants.java]: Small change to comment.
```suggestion
  public static int SLOTS_PER_HISTORICAL_ROOT = 8192; // 2^13 slots (~13 hours)
```

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/Constants.java [hunk 2]
```diff
@@ -48,7 +48,9 @@
...
   public static int MIN_VALIDATOR_WITHDRAWABILITY_DELAY = 256; // 2^8 epochs (~27 hours)
+  public static int PERSISTENT_COMMITTEE_PERIOD = 2048; // 2^8 epochs (~27 hours)
```
Comment (schroedingerscode) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/Constants.java]: Small change to comment.
```suggestion
  public static int PERSISTENT_COMMITTEE_PERIOD = 2048; // 2^11 epochs (~9 days)
```

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java [hunk 3]
```diff
@@ -281,21 +318,27 @@ public Bytes toBytes() {
...
+          writer.writeBytesList(latest_state_roots);
           writer.writeBytesList(latest_active_index_roots);
           writer.writeULongIntList(
               64,
               latest_slashed_balances.stream()
                   .map(UnsignedLong::longValue)
                   .collect(Collectors.toList()));
-          writer.writeBytesList(latest_attestationBytes);
-          writer.writeBytesList(batched_block_roots);
+          writer.writeBytesList(latest_block_header.toBytes());
```
Comment (schroedingerscode) [inline ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java]: latest_block_header is not a list.
```suggestion
          writer.writeBytes(latest_block_header.toBytes());
```

**Review by schroedingerscode — CHANGES_REQUESTED** [review]
One serialization issue to be addressed, and a handful of minor documentation issues.

File: ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java (hunk 3 above)
Comment (cemozerr) [reply ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/state/BeaconState.java]: Great catch.

**Review by schroedingerscode — APPROVED** [review]
LGTM

**Pull request merged**